### ISSUE TITLE
[Suggestion] /siege alternative (From GriefPrevention)

### ISSUE THREAD

**Issue by FrozenBeard** [bloodmc/GriefDefender#47]
Hello! 

Im looking for an alternative to GriefPrevention that is updated more regular and is possible to request features too, GP has been slacking for a very long time now.
The one feature missing from GriefPrevention  -> GriefDefender is the ability to siege.
Siege has always sucked with GriefPrevention, Ive used it for 5/6 years now and its gone through multiple developers hands each with a different take on siege.

Im not sure if this would be better as an addon plugin for GriefDefender.

One of the biggest issues with siege, is there is no log, statistics, or records of peoples attacking/defending, or how to group these people together to understand, limit and record attacks/defends, If siege were to happen with GriefDefender I think it would need:

1. A statistics system that records all battles for won/loss of a siege and those involved in it to build up a reputation score per player, this allows stopping a high level player attacking a brand new player for example.
2. Players working together in a Crew/Clan, We use SimpleClans, its brilliant and its got an API, this would solve all whos working with who, its already got a k/d system too. Expanding from this would be brilliant, its also got a /crew war feature thats not triggering anything at present but can be hooked into when 2 crews rival each other (they both have to accept the rivalry)
This could be the best solution for starting off with a new siege plugin as both parties have to accept to be rivals.

Very basic overview, I don't want to get too much into detail just incase you don't want to ever add a siege system, then I will be praying on Griefprevention updates!

Thanks

**Comment by bloodmc** [comment]
There will be no siege addon to GriefDefender as it is out of scope for what this plugin is about which is protection. However, I will be adding support for Guilds soon which can be found here
https://www.spigotmc.org/resources/guilds-new-mysql-sqlite-support-update-sale-20-off.66176/

**Comment by FrozenBeard** [comment]
Never seen that Guilds, looks like its got some cool features with the wars and vault, sadly its missing so much compared to SimpleClans, loads of really basic necessities for running a crew. I guess its a fairly new plugin, SimpleClans has been around forever, Just so you are aware the SimpleClans Bukkit/Spigot pages are not updated its all happening in the GitHub.
https://github.com/marcelo-mason/SimpleClans/releases

**Comment by FrozenBeard** [comment]
Ahhh right, the old fliparoo, Its always funny asking for a developer to add hooks into others APIs, they usually say "go to the other person to add it" then they are like "get them to add it" when they both have API's haha 

I will ask him! I'm currently using GriefPrevention and haven't tested GD yet, quite a big change for us, sitting at 27,329 claims, and did already ask him to add a hook into GP but it never came about. 

SimpleClans really is brilliant for a fully fledged crew management plugin, it focuses on just that managing a crew not anything with land claims, thats what would make it utterly perfect to combine the two GriefDefender & SimpleClans, using the Simpleclans rivalry, k/d etc and war system to activate fights in a claim, I will talk with the SimpleClans dev see what he thinks, its just hard when ive not used GriefDefender to think of exactly how it should work.

I really need to learn Java so I can create my own man in the middle plugin to work between plugins, it would mean figuring out how a siege system would work though.

Your https://github.com/GriefDefenderAPI leads to a 404, is this it? https://github.com/bloodmc/GriefDefenderAPI

**Comment by FrozenBeard** [comment]
Lets start super basic, SimpleClans creates a permissions for each clan (Apparently) which would lead me to believe it should be simple enough to trust an entire Clan into a claim plugin /trust clan or something, as each person in the Clan *should* inherit their clans permission they should trust/container etc trust to the Claim it, being kicked from a Clan means they are auto ejected from the Claim then.

Its called SimpleClans but they allow you to change all the commands in-game which not many plugins so, so my entire system has been re-branded to Crews being a Pirate themed server, so may end up calling Clans Crews.

Once this exists; then we can progress into Alliances/Rivalries & wars!

**Comment by bloodmc** [comment]
OK so to distinguish between normal permission group trust and clan trust, I'm thinking a new command such as `/trustclan` or `/clantrust` which would add a prefix `clan-` before name. 

So for ex. if a player typed `/trustclan viper`, GD would add `clan-viper` to builder group. 
If the same user typed `/at viper` , GD would add `clan-viper` to access trust group. 

Thoughts?

**Comment by FrozenBeard** [comment]
Wouldn't access trust need to be the same? `/atcrew viper` An alternative could be:
`/trust crew viper`
`/at crew viper`
Just means no player can be named Crew, be a good idea to have the clan/crew variable changeable per server, I would use crew.

Either that or
```/trustcrew viper
/atcrew viper
```
But not permission trust, that would get a bit mental!

**Comment by LukeJS** [comment]
Why not something like `/trust clan:viper` or `/trust c:viper`?

**Comment by bloodmc** [comment]
Yes, I had a typo. OK so basically we agree that 2 commands should be added

`/trustclan|trustcrew [clanname]`
`/atclan|trustclan [clanname]`

So you are free to use clan or crew as I'll have both added as aliases.

**Comment by FrozenBeard** [comment]
Sounds good to me!

**Comment by color-space** [comment]
Second this! 
I really would appreciate clans aswell
